src/metric.jl

```diff
@@ -235,31 +221,34 @@ Calculates the mean squared error regression loss.
 Requires that label and prediction have the same shape.
 """
 
-mutable struct MSE <: AbstractEvalMetric
-  mse_sum  :: Vector{NDArray}
+mutable struct MSE{T<:NDArray} <: AbstractEvalMetric
+  mse_sum  :: Vector{T}
```

Comment:
@vchuravy I changed `MSE` like this, but I got confused that how does user initialize a `MSE` instance?
In the example of regression, it just creates a instance via `MSE()`.
And...what's your design of struct `MSE` if we have parametric `NDArray`?

Reply:
Since this is for error reporting only it should simply default to `Float64` or `Float32`.
The array is simply to not block during `_update_single_output`.

Reply:
> Since this is for error reporting only it should simply default to `Float64` or `Float32`.

okay, I will init them with `MX_float` (`Float32`) by default.

> The array is simply to not block during `_update_single_output`.

I do not get what you mean.